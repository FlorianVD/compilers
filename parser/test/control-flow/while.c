int whiles()
{
    while (1 == 2) {
        42;
    }

    while (1 == 2) ;
}
