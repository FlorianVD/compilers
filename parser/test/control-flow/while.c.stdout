└── Program
    └── FuncDecl: returnType = 'int', name = 'whiles'
        └── CompoundStmt
            ├── WhileStmt
            │   ├── BinaryOpExpr: op = '=='
            │   │   ├── IntLiteral: value = '1'
            │   │   └── IntLiteral: value = '2'
            │   └── CompoundStmt
            │       └── ExprStmt
            │           └── IntLiteral: value = '42'
            └── WhileStmt
                ├── BinaryOpExpr: op = '=='
                │   ├── IntLiteral: value = '1'
                │   └── IntLiteral: value = '2'
                └── EmptyStmt
